x-pack/plugins/maps/public/classes/layers/tiled_vector_layer/tiled_vector_layer.tsx

```diff
@@ -182,7 +182,10 @@ export class TiledVectorLayer extends VectorLayer {
     const layerIds = this.getMbLayerIds();
     for (let i = 0; i < layerIds.length; i++) {
       const mbLayer = mbMap.getLayer(layerIds[i]);
-      if (mbLayer && mbLayer['source-layer'] !== tiledSourceMeta.layerName) {
+      // The mapbox type in the spec is specified with `source-layer`
+      // but the programmable JS-object uses camelcase `sourceLayer`
+      // @ts-expect-error
```

Comment:
Mind creating a PR against https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/mapbox-gl to fix https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/mapbox-gl/index.d.ts#L1891?